pkg/apis/crd/v1beta1/types.go

```diff
@@ -224,6 +226,16 @@ type IPRange struct {
 	End string `json:"end,omitempty"`
 }
 
+// SubnetInfo specifies subnet attributes for IP Range.
+type SubnetInfo struct {
+	// Gateway IP for this subnet, e.g. 10.10.1.1.
+	Gateway string `json:"gateway"`
+	// Prefix length for the subnet, e.g. 24.
+	PrefixLength int32 `json:"prefixLength"`
```

Comment:
I was wondering if it would be more natural for people to explicitly have to provide a CIDR for the subnet and a separate IP address (part of the CIDR) for the gateway, as opposed to deducing the subnet from the gateway IP + prefix length. This may be more in line with how people are used to interact with `ip` tools? But I don't have a strong opinion.

Reply:
I refered  to SubnetInfo in `IPPool` for this, though not completely. Let me check how the struct was determined and how such configuration was provided in other systems.

Reply:
It seems when `IPPool` was designed, it just asked the least information from users to construct a subnet. Asking a subnet may be a bit tedious for users as they may have provided the same information in `ipRanges` once. For example, the 2nd config below would look strange, and we need have one more validation to ensure the provided subnet matches the ip ranges.
```
  ipRanges:
  - cidr: 10.10.0.0/24
  subnetInfo:
    gateway: 10.10.0.1
    prefixLength: 24
    vlan: 10
```

```
  ipRanges:
  - cidr: 10.10.0.0/24
  subnetInfo:
    gateway: 10.10.0.1
    subnet: 10.10.0.0/24
    vlan: 10
```